rotkehlchen/db/history_events.py

```diff
@@ -500,10 +502,17 @@ def get_history_events(
                     deserialized_event = EthDepositEvent.deserialize_from_db(data)
 
                 else:
-                    data = entry[data_start_idx:HISTORY_BASE_ENTRY_LENGTH + 1]
-                    deserialized_event = HistoryEvent.deserialize_from_db(entry[data_start_idx:])
+                    data = entry[data_start_idx:]
+                    deserialized_event = HistoryEvent.deserialize_from_db(data)
             except (DeserializationError, UnknownAsset) as e:
-                log.debug(f'Failed to deserialize history event {entry} due to {e!s}')
+                log.error(f'Failed to deserialize history event {entry} due to {e!s}')
+                if msg_aggregator is not None:
```

Comment:
hmmm this is wrong. You don't need to pass it as optional and check it in here. You already have it.

`self.database.msg_aggregator`